rfcs/038-sierra-series/README.md

```diff
@@ -0,0 +1,432 @@
+  {
+    "id": "cpp2xh99",
+    "title": "Articles and Papers on Nutrition",
+    "referenceNumber": "PP/DSM/3/4",
+    "totalParts": 0,
+    "totalDescendentParts": 0,
+    "type": "Work"
+  },
+  {
+    "id": "ur5sakg9",
+    "title": "\"Nutrition in Britain in the twentieth century\" - PhD thesis",
+    "referenceNumber": "PP/DSM/3/5",
+    "totalParts": 0,
+    "totalDescendentParts": 0,
+    "type": "Work"
+  },
+  {
+    "id": "k4vp98bp",
+    "title": "Miscellaneous Items",
+    "referenceNumber": "PP/DSM/3/6",
+    "totalParts": 0,
+    "totalDescendentParts": 0,
+    "type": "Work"
+  }
+],
+"partOf": [
+  {
+    "id": "hb9hqd8p",
+    "title": "McLaren, Dr Donald S",
+    "referenceNumber": "PP/DSM",
+    "partOf": [],
+    "totalParts": 3,
+    "totalDescendentParts": 9,
+    "type": "Collection"
+  }
+],
+```
+For archive collections, the website renders these as a collapsible hierarchy: 
+
+![](archives.png)
+
+Given the size of many series and Host Item Entry that would not be practical in this case, so the proposal 
+is for the frontend to render differently the information in `partOf` based on the `type` property of the work. 
+
+In the case of archive works, the frontend should render the hierarchy as in the example above.
+
+In the case of Sierra works, the frontend should provide a link to a search page where the user can paginate through 
+all works linked by the same Series or Host Item Entry. 
+To do this the Catalogue API should provide a filter that matches exactly on the contents of `partOf` along the lines of
+`/works?partOf.title=Morphogenesis of the vertebrate brain`.
+
+As series and Host Item Entries can have volumes, the proposal is to represent them by creating one record for the Series or
+Host Item Entry and one for the volume with a nested `partOf` property (see example below).
+
+Additionally, if a series or Host Item Entry has an identifier (such as ISSN os ISBn or other) the proposal is 
+to mint a canonical identifier in the pipeline and to expose that identifier in the API. 
+Finally, the proposal is to model a series statement or a Host Item Entry with a type `Work`.
+
+### Examples
+ 
+- [1074785](https://search.wellcomelibrary.org/iii/encore/record/C__Rb1074785?lang=eng&suite=cobalt&marcData=Y) 
+This is an example of a series statement in 490 + 830. In this example the information contained in 830 and 490 is 
+  exactly the same, so the result is just one statement. 
+```yaml
+"partOf": [
+  {
+    "title": "Morphogenesis of the vertebrate brain; 2",
+    "workType": {
+      "id": "???",
+      "label": "Volumes",
+      "type": "Format"             
+    },
+    "partOf": [
+      {
+        "title": "Morphogenesis of the vertebrate brain",
+        "type": "Series"
+      }
+    ],
+    "type": "Series"
+  }
+]
+```
+- [3109805](https://search.wellcomelibrary.org/iii/encore/record/C__Rb3109805?lang=eng&suite=cobalt&marcData=Y)
+  490 + 773 with slightly different information
+```yaml
+"partOf": [
```

Comment:
This example neatly contradicts what I just said above, as it does look more clearly hierarchical.

Indeed in this instance it would be most correct to have "Early European Books : Printed sources to 170" as partOf "Early European Books" (ie the 773).

If that's too difficult, it could also be list that is de-duped across the values generated from 490/830 and 773. It's likely to rendered as a list on the front end anyway.

Reply:
I think the rules to implement in the transformer to figure out when a host item entry field completely matches a series statement, when it's a more general higher level and when it's not related at all could become quite complicated. In these examples I tried to go for: if it matches completely a series statement -> collapse; if it doesn't -> keep it separate. The main thinking is that this is easy to implement and limits the chance that we present information that is wrong. 
I think it might be difficult to come up with a rule that could figure out in this instance what case this should fall into, so I would just go for not trying to be too clever at least as a first iteration and just accept that on the frontend there will be a list and that some elements on that list will appear as duplicates to a human

Reply:
Agreed, let's de-duplicate an exact match. We can then keep an eye out for things that need data cleanup and also any examples that we need to look at further.